src/transformers/trainer.py

```diff
@@ -2499,37 +2495,9 @@ def push_to_hub(
         if not self.is_world_process_zero():
             return
 
-        if not isinstance(unwrap_model(self.model), PushToHubMixin):
-            raise ValueError(
-                "The `upload_model_to_hub` method only works for models that inherit from `PushToHubMixin` models."
-            )
-
-        if repo_url is None and repo_name is None:
-            repo_name = Path(self.args.output_dir).name
-
-        if repo_name is not None:
-            model_name = repo_name
-        elif repo_url is not None:
-            model_name = repo_url.split("/")[-1]
-        else:
-            model_name = None
-        self.create_model_card(model_name=model_name, **kwargs)
-
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            shutil.copy(os.path.join(self.args.output_dir, "README.md"), os.path.join(tmp_dir, "README.md"))
-            unwrap_model(self.model).save_pretrained(tmp_dir)
-            if self.tokenizer is not None:
-                self.tokenizer.save_pretrained(tmp_dir)
-
-            return unwrap_model(self.model)._push_to_hub(
-                save_directory=tmp_dir,
-                repo_name=repo_name,
-                repo_url=repo_url,
-                commit_message=commit_message,
-                organization=organization,
-                private=private,
-                use_auth_token=use_auth_token,
-            )
+        self.create_model_card(model_name=self.args.push_to_hub_model_id, **kwargs)
+        self.save_model()
+        return PushToHubMixin._push_to_hub(repo=self.repo, commit_message=commit_message)
```

Comment:
```suggestion
        return self.repo.push_to_hub(commit_message=commit_message)
```

maybe just (no strong opinion)

Reply:
Yes it's the exact same thing since `_push_to_hub` just adds a default for the commit message.
So this is cleaner like this!